src/main/java/fundamentals/MyLinkedList.java

```diff
@@ -0,0 +1,162 @@
+package fundamentals;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MyLinkedList<T> {
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+        this.first = null;
+        this.last = null;
+    }
+
+    public void addFirst(final T data) {
+        Node<T> newNode = new Node<T>() {
```

Comment:
This is wrong on multiple layers. First abstraction layers are mixed. Second this tightly couples implementation of `Node` with list add method.

Reply:
corrected